infrastructure/gcp/keep-test/google-functions/keep-faucet/issue-grant.js

```diff
@@ -126,12 +155,43 @@ exports.issueGrant = async (request, response) => {
               You can find us on Discord at: https://discord.gg/jqxBU4m .\n
             `)
           })
-          .on('error', (error) => {
-            console.error(
-              `Error with account grant transaction: [${error}]; URL was [${request.url}].`
-            )
-            if (!response.headersSent) {
-              response.status(500).send(`
+          .on("error", (error) => {
+            if (error == "Error: nonce too low") {
+              const noncePlusOne = nonce + 1
+              console.error(
+                `Error with account grant transaction, Nonce too low: At [${nonce}], retry at [${noncePlusOne}]; URL was [${request.url}].`
+              )
+              console.log("Resubmitting transaction with higher nonce...")
```

Comment:
A cleaner implementation here might be to give `issueGrant` an additional parameter, `abortOnFailure`. If it's set to `true`, `on("error"`, send the 500. If not, `on("error"` you can call `exports.issueGrant(request, response, true)`. That lets you save the repetition of this code, and can be adapted later to >1 retry if needed.